kedro/extras/datasets/tensorflow/tensorflow_model_data_set.py

```diff
@@ -0,0 +1,166 @@
+            fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """
+        Args:
+            filepath: Filepath to a TensorFlow model directory prefixed with a protocol like `s3://`.
+                If prefix is not provided `file` protocol (local filesystem) will be used.
+                The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            load_args: TensorFlow options for loading models.
+                custom_objects: Optional dictionary mapping names (strings) to custom classes or functions to be
+                considered during deserialization.
+                compile: Boolean, whether to compile the model after loading.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/load_model
+                All defaults are preserved.
+            save_args: TensorFlow options for saving models.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/save_model
+                All defaults are preserved, except for "index", which is set to False.
+                In TF this defaults to 'tf' in TF 2.X, and 'h5' in TF 1.X.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{'token': None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class.
+                E.g. for ``GCSFileSystem`` class: `{project: 'my-project', ...}`
+        """
+        _fs_args = copy.deepcopy(fs_args) or {}
+        _credentials = copy.deepcopy(credentials) or {}
+        protocol, path = get_protocol_and_path(filepath, version)
+
+        self._protocol = protocol
+        self._fs = fsspec.filesystem(self._protocol, **_credentials, **_fs_args)
+        super().__init__(
+            PurePath(filepath),
+            version,
+            # exists_function=lambda x: Path(x).is_dir(),
+            # exists_function=self._fs.isdir,
+            exists_function=self._fs.exists,
+        )
+
+        self._data = None
+
+        # Handle default load and save arguments
+        self._load_args = copy.deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+    def _load(self) -> None:
+        load_path = Path(self._get_load_path())
+        return tf.keras.models.load_model(str(load_path), **self._load_args)
```

Comment:
I'm a little stuck here, because we should be opening `load_path` using `fsspec` and then passing the file descriptor to `load_model` so that users can use any filesystem they want and aren't tied to local, but it looks like `load_model` only wants a string or an `h5py.File` object? Is there anything we can do here?